docs/migration-guides/v2/lib.md

```diff
@@ -0,0 +1,284 @@
+- The ability to use a boolean value for the `recover` client option has been removed in v2. If you wish for the connection to always be recovered, replace `{ recover: true }` with a function that always passes `true` to its callback: `{ recover: function(lastConnectionDetails, cb) { cb(true); } }`.
+- The `log` client option has been removed in v2. Equivalent functionality is provided by the `logLevel` and `logHandler` client options. Update your client options code of the form `{ log: { level: logLevel, handler: logHandler } }` to instead be `{ logLevel, logHandler }`.
+- The ability to pass an array of channel mode flags as the first argument of `RealtimeChannel.attach()` has been removed in v2. To set channel mode flags, populate the `modes` property of the channel options object that you pass to `Channels.get()` or `RealtimeChannel.setOptions()`.
+- The `force` auth option has been removed in v2. If you’re using this option to force `authorize()` to fetch a new token even if the current token has not expired, this is no longer necessary, as `authorize()` now always fetches a new token. Update your code to no longer pass the `force` auth option. Note that, in general, passing an auth options argument to `authorize()` will overwrite the library’s stored auth options, which may not be what you want. In v1, the library contains a special case behavior where passing an auth options object which only contains `{ force: true }` will _not_ overwrite the stored options. This special case behavior has been removed in v2, so if you’re currently passing `authorize()` an auth options object which only contains `{ force: true }`, you should stop passing it an auth options object entirely.
+- The `host` client option has been renamed to `restHost`. Update your code to use `restHost`.
+- The `wsHost` client option has been renamed to `realtimeHost`. Update your code to use `realtimeHost`.
+- The `queueEvents` client option has been renamed to `queueMessages`. Update your code to use `queueMessages`.
+- `RealtimePresence`’s `on` method has been renamed to `subscribe`. Update your code to use `subscribe`.
+- `RealtimePresence`’s `off` method has been renamed to `unsubscribe`. Update your code to use `unsubscribe`.
+- `Auth`’s `authorise` method has been renamed to `authorize`. Update your code to use `authorize`.
+- The `"xhr"` transport has been renamed to `"xhr_streaming"`. Update your client options code to use `transports: ["xhr_streaming"]`.
```

Comment:
Yep, depending on which PR gets first either @owenpearson will need to update the migration guide in his PR (see https://github.com/ably/ably-js/pull/1645#issuecomment-2004963270), or I'll update it in this PR.

Reply:
just a heads up - i'm gonna merge that PR as soon as CI is all passing on it. still happy to contribute any changelog stuff needed though whether that be just as a comment here or as a commit on this branch

Reply:
I see it's been merged now. As far as I'm aware, the only user-facing change — at least that would need to be mentioned in a migration guide — is the removal of the `xhr_streaming` transport, is that right? So it's just modifying this line to Andrii’s suggested text above?

Reply:
I think we should put something more descriptive about how the transport connection sequence has changed too, since there will be a regression for anyone using xhr_polling, in that the initial connection will be slower until transport preference has been cached in the browser, although we should also make it clear that it's an overall improvement for anyone who isn't using xhr_polling by focussing on the fact that we no longer initially connect with comet and then try to upgrade, instead we always try web_socket first if it's available.

Reply:
I'm not familiar with what was implemented in #1645 so I'll put some guidance in that's a paraphrase of what you've written above, then you can tell me if it looks right :)

Reply:
Also what I said in https://github.com/ably/ably-js/pull/1670#discussion_r1530755508 is wrong; this needs to be moved from the "Stop using functionality that v1 deprecated and which v2 removes" section to the "Stop using other functionality that v2 removes" section, will do.

Reply:
Actually @owenpearson would you mind writing something that we can add to describe these changes? I think it'd be a lot quicker than me trying to understand it. Feel free to push a commit.

Reply:
I've added it as a suggestion here https://github.com/ably/ably-js/pull/1670#discussion_r1532206072, i'm not super happy with the long paragraphs and placement within the document so feel free to edit as much as you like, hopefully that's at least a helpful starting point.

Reply:
Thanks, have:

- very lightly edited what you wrote in that comment there and added it
- added section re removal of `xhr` / `xhr_streaming`

Reply:
I believe this is now resolved, so @lawrence-forooghian feel free to close this conversation if you've added everything you wanted with Owen.